spec.emu

```diff
@@ -13,22 +13,73 @@ contributors:
   - Leo Balter
   - Caridy Patiño
   - Rick Waldron
+  - Ashley Claymore
 </pre>
 
-<emu-clause id="sec-isobjectorsymbol" aoid="IsObjectOrSymbol">
-  <h1>IsObjectOrSymbol ( _argument_ )</h1>
-  <p>The abstract operation IsObjectOrSymbol takes argument _argument_ (an ECMAScript language value) and determines if its Type is Object or Symbol. This operation performs the following steps when called:</p>
+<emu-clause id="sec-ecmascript-data-types-and-values" aoid="Type">
+  <h1>ECMAScript Data Types and Values</h1>
+  <emu-clause id="sec-object-type">
+    <h1>The Object Type</h1>
+    <emu-clause id="sec-well-known-intrinsic-objects">
+      <h1>Well-Known Intrinsic Objects</h1>
+      <emu-table id="table-well-known-intrinsic-objects" caption="Well-Known Intrinsic Objects" oldids="table-7">
+        <table>
+          <tr>
+            <th>
+              Intrinsic Name
+            </th>
+            <th>
+              Global Name
+            </th>
+            <th>
+              ECMAScript Language Association
+            </th>
+          </tr>
+          <tr>
+            <td>
+              <ins>%CanBeHeldWeakly%</ins>
+            </td>
+            <td>
+            </td>
+            <td>
+              <ins>A function to check if a value can be held weakly (<emu-xref href="#sec-%canbeheldweakly%"></emu-xref>)</ins>
+            </td>
+          </tr>
+        </table>
+      </emu-table>
+    </emu-clause>
+  </emu-clause>
+</emu-clause>
+<emu-clause id="sec-%canbeheldweakly%">
+  <h1>%CanBeHeldWeakly%( _argument_ )</h1>
+  <p>The <dfn>%CanBeHeldWeakly%</dfn> intrinsic is an anonymous built-in function object that is defined once for each realm. When %CanBeHeldWeakly% is called with one argument _argument_ (an ECMAScript language value) the following steps are taken:</p>
   <emu-alg>
-    1. If Type(_argument_) is Object or Symbol, return *true*.
+    1. If Type(_argument_) is Object, return *true*.
+    1. If Type(_argument_) is Symbol, then
+      1. For each element _e_ of the GlobalSymbolRegistry List (see <emu-xref href="#sec-symbol.for"></emu-xref>), do
+        1. If SameValue(_e_.[[Symbol]], _argument_) is *true*, return *false*.
+      1. Let _wellKnownSymbols_ be the symbols listed in <emu-xref href="#sec-well-known-symbols"></emu-xref>.
+      1. If _wellKnownSymbols_ contains _argument_, return *false*.
```

Comment:
This might be relevant here: https://github.com/tc39/test262/issues/3420.

The %Intl%.[[Fallback]] symbol is specified as a unique symbol associated with the realm. The text here would consider it to be unforgeable, which is theoretically true and will be in some implementations, e.g. Spidermonkey.

V8 however implemented the fallback symbol with the single-identity-across-realms characteristic of a well-known symbol. Though technically incorrect, it seems this may have been a deliberate decision and the discussion about which behavior is desired did not appear conclusive. This proposal might force the issue of getting engines / the Intl spec to agree on the true nature of this abomination.

Reply:
Thats something that needs to either be changed in the spec, or, v8 needs to change their implementation, and it should be discussed in plenary regardless of this PR/proposal.

Reply:
If I have understood that issue, this is not an issue for this proposal in practice. Because the symbol value itself is not exposed anywhere, so it wouldn't be possible to try and use it in a `WeakSet` for example. So the spec would say that it is not well-known (cross realm) so _can_ be used in a `WeakSet`, V8 implement it cross-realm so it is well-known so if they didn't allow it in a `WeakSet` that would be incorrect, but that incorrect behaviour can't be triggered in any way.

Please do correct anything I have misunderstood.

Reply:
Yes. Though as you mentioned in another thread, the status of well-knowns as “forgeable” is pretty iffy\*. If it’s decided that well-known symbols aren’t considered forgeable, then this outlier-in-practice thing is moot anyway.

\* It’s strictly impossible to “prove” either conclusion in all possible scenarios from an ES code runtime POV, so while the spec describes them in terms that consistently support the unforgeable conclusion, I understand that this isn’t cut and dry since if a tree falls in the forest...